src/components/shared/OrderSelector/index.tsx

```diff
@@ -0,0 +1,114 @@
+import { useTranslation } from 'react-i18next';
+
+import {
+  FormControl,
+  Select,
+  MenuItem,
+  SelectChangeEvent,
+} from '@mui/material';
+
+import theme from 'src/theme';
+
+export type SortParameter = 'price' | 'name' | 'createdAt';
+export type SortOrder = 'ASC' | 'DESC';
+
+interface SortProps {
+  sortBy?: SortParameter;
+  sortOrder?: SortOrder;
+  onSortChange: (sortBy?: SortParameter, sortOrder?: SortOrder) => void;
+}
+
+const options = [
+  { value: 'none', labelKey: 'orderSelect.noSort' },
+  {
+    value: 'price:ASC',
+    labelKey: 'orderSelect.price',
+    orderKey: 'orderSelect.ascending',
+  },
+  {
+    value: 'price:DESC',
+    labelKey: 'orderSelect.price',
+    orderKey: 'orderSelect.descending',
+  },
+  {
+    value: 'createdAt:ASC',
+    labelKey: 'orderSelect.date',
+    orderKey: 'orderSelect.ascending',
+  },
+  {
+    value: 'createdAt:DESC',
+    labelKey: 'orderSelect.date',
+    orderKey: 'orderSelect.descending',
+  },
+  {
+    value: 'name:ASC',
+    labelKey: 'orderSelect.name',
+    orderKey: 'orderSelect.ascending',
+  },
+  {
+    value: 'name:DESC',
+    labelKey: 'orderSelect.name',
+    orderKey: 'orderSelect.descending',
+  },
+];
+
+function OrderSelector({ sortBy, sortOrder, onSortChange }: SortProps) {
+  const { t } = useTranslation();
+
+  const handleSortChange = (event: SelectChangeEvent<string>) => {
+    const { value } = event.target;
+    const newSortBy =
+      value === 'none' ? undefined : (value.split(':')[0] as SortParameter);
+    const newSortOrder =
+      value === 'none' ? undefined : (value.split(':')[1] as SortOrder);
```

Comment:
find other solution, if you use AS its mean that something wrong

Reply:
fixed